client/src/components/settings/Certification.test.js

```diff
@@ -15,75 +15,62 @@ describe('<certification />', () => {
   // shallow rendering does not render children component
   // form buttons are not included in shallow render
   it('Should render show cert button for claimed legacy cert', () => {
-    const { container } = renderWithRedux(
-      <CertificationSettings {...defaultTestProps} />
-    );
+    renderWithRedux(<CertificationSettings {...defaultTestProps} />);
 
     expect(
-      container.querySelector(
-        'a[href="/certification/developmentuser/legacy-data-visualization"]'
-      )
-    ).toHaveTextContent('buttons.show-cert');
-  });
-
-  it('Should link show cert button to the claimed legacy cert', () => {
-    const { container } = renderWithRedux(
-      <CertificationSettings {...defaultTestProps} />
+      screen.getByRole('link', {
+        name: 'buttons.show-cert'
+      })
+    ).toHaveAttribute(
+      'href',
+      '/certification/developmentuser/legacy-data-visualization'
     );
-
-    expect(
-      container.querySelector(
-        'a[href="/certification/developmentuser/legacy-data-visualization"]'
-      )
-    ).toBeInTheDocument();
   });
 
   // full forms with unclaimed certs should not shallow render show cert button
   it('Should not render show cert button for unclaimed cert with completed projects', () => {
-    const { container } = renderWithRedux(
-      <CertificationSettings {...defaultTestProps} />
-    );
+    renderWithRedux(<CertificationSettings {...defaultTestProps} />);
 
-    expect(
-      container.querySelector(
-        'a[href="/certification/developmentuser/legacy-back-end"]'
-      )
-    ).not.toHaveTextContent('buttons.show-cert');
+    const allUnclaimedCerts = screen.getAllByRole('link', {
+      name: 'buttons.claim-cert'
+    });
+    const unclaimedCompletedProject = allUnclaimedCerts.filter(
+      element =>
+        element.getAttribute('href') ===
+        '/certification/developmentuser/legacy-back-end'
+    );
+    expect(unclaimedCompletedProject).toHaveLength(1);
   });
 
   // empty forms with unclaimed certs should not shallow render show cert button
   it('Should not render show cert button for cert with no completed projects', () => {
-    const { container } = renderWithRedux(
-      <CertificationSettings {...defaultTestProps} />
-    );
+    renderWithRedux(<CertificationSettings {...defaultTestProps} />);
 
-    expect(
-      container.querySelector(
-        'a[href="/certification/developmentuser/legacy-front-end"]'
-      )
-    ).not.toHaveTextContent('buttons.show-cert');
+    const allUnclaimedCerts = screen.getAllByRole('link', {
+      name: 'buttons.claim-cert'
+    });
+    const unclaimedIncompleteProject = allUnclaimedCerts.filter(
+      element =>
+        element.getAttribute('href') ===
+        '/certification/developmentuser/legacy-front-end'
+    );
+    expect(unclaimedIncompleteProject).toHaveLength(1);
```

Comment:
I am bit skeptical of this, ie, if is there a better/more correct way to test the same given the new constraint we have.. Suggestions  are welcome.. 😄

Reply:
Thanks @huyenltnguyen For the suggestion.. I have refractored and implemented your suggestions..